aggregator/src/main.rs

```diff
@@ -0,0 +1,84 @@
+use crate::{
+    common::{constants::MBUS_URL, error},
+    events::events_store::initialize,
+};
+
+use k8s_openapi::api::core::v1::ConfigMap;
+mod common;
+mod events;
+use events::events_cache::{Cache, EventSet};
+use mbus_api::mbus_nats::{message_bus_init, NatsMessageBus};
+use tracing::{error, info};
+use utils::{
+    raw_version_str,
+    tracing_telemetry::{default_tracing_tags, flush_traces, init_tracing},
+};
+
+async fn mbus_init() -> error::Result<NatsMessageBus> {
+    let message_bus = message_bus_init(MBUS_URL).await;
+    Ok(message_bus)
+}
+
+/// Initialize events store cache from config map.
+async fn initialize_events_cache(init_data: ConfigMap) {
+    let events = EventSet::from_event_store(init_data).unwrap();
+    Cache::initialize(events);
+}
+
+/// Initialize events store.
+async fn initialize_events_store(namespace: &str) -> error::Result<ConfigMap> {
+    let event_store = initialize(namespace).await?;
+    Ok(event_store)
+}
+
+#[tokio::main]
+async fn main() -> error::Result<()> {
+    init_logging();
+    info!("stats aggregation started!");
+
+    let m_bus = mbus_init().await?;
+    info!("mbus initialized successfully!");
+
+    let init_data = initialize_events_store("mayastor").await?;
+    info!("event store initialized successfully!");
+    
+    initialize_events_cache(init_data).await;
+    info!("event cache initialized successfully!");
+
+    // spawn a new task to store the data in cache.
+    tokio::spawn(async move {
+        events::events_cache::store_events(m_bus)
+            .await
+            .map_err(|error| {
+                error!(%error, "Error while storing the events to cahce");
+                flush_traces();
+                error
+            })
+    });
+
+    // spawn a new task to update the config map from cache.
+
+    tokio::spawn(async move {
+        events::events_store::update_config_map_data("mayastor")
+            .await
+            .map_err(|error| {
+                error!(%error, "Error while persisting the stats to config map from cache");
+                flush_traces();
+                error
+            })
+    });
+
+    for n in 1..100 {
+        info!("Inside for loop {}th time!", n);
+        tokio::time::sleep(std::time::Duration::from_secs(30)).await;
+    }
+
+    Ok(())
+}
```

Comment:
IMO, there should be a task listening for SIGINT, SIGTERM and it should try to flush the aggregated events to the bus, and this should be fallible, as on uninstall the bus might have been removed earlier. Kubernetes Pods have a default grace period of 30 seconds, we can maybe aslo retry a couple of times before giving up. It should be fallible and exit with success.

Reply:
as per the design , this is not present. Can be taken later if agreed. for now , the mvp must be in

Reply:
Sure..